eeps/eep-0072.md

```diff
@@ -0,0 +1,227 @@
+    Author: Jesse Gumm <ja(at)gumm(dot)io>
+    Status: Draft
+    Type: Standards Track
+    Created: 08-Oct-2024
+    Post-History:
+****
+EEP 72: Reserved words and Variables as record names, and enhancement to definition syntax
+----
+
+Abstract
+========
+
+This EEP loosens some of the restrictions around record names to make it
+no longer necessary to quote them when they are named with reserved
+words (`#if` vs `#'if'`) or words with capitalized first characters (`#Hello`
+vs `#'Hello'`).
+
+This EEP also proposes to add a new record-like syntax to the record
+definitions (also adopting the above syntactical changes), so that the
+following record definitions would all be valid and identical:
+
+```erlang
+-record('div', {a :: integer(), b :: integer()}).
+-record #'div'{a :: integer(), b :: integer()}.
+-record #div{a :: integer(), b :: integer()}.
+```
+
+The latter two are proposed new syntax.  The following would also be valid
+and identical since parentheses are optional in attributes:
+
+```erlang
+-record 'div', {a :: integer(), b :: integer()}.
+-record(#'div'{a :: integer(), b :: integer()}).
+-record(#div{a :: integer(), b :: integer()}).
+```
```

Comment:
I think the intention to add *one* new format would be better shown by just having `-record('div', {` and `-record #div{` in the first group, and write that due to optional quoting and optional parentheses the second group (of 4) are identical and equivalent.

Reply:
Yes, that's right - I forgot to adjust this in my round of updates.  I will adjust accordingly.